### ISSUE TITLE
reset_system_mode error

### ISSUE THREAD

**Issue by lloyda** [zxdavb/evohome#63]
Calling `ramses_cc.reset_system_mode` from HA is not working in v**0.22.1**

Example HA automation:

```yaml
- id: '1607967971509'
  alias: Reset schedule on Sunday
  description: Resets the heating system to follow schedule at 13:30 each Sunday
  trigger:
  - platform: time
    at: '13:20'
  condition:
  - condition: time
    weekday:
    - sun
  - condition: state
    entity_id: climate.controller
    attribute: system_mode
    state: Auto
  action:
  - service: ramses_cc.reset_system_mode
    data:
      entity_id: climate.controller
  mode: single
```
Error in HA log:

```
2022-11-14 15:52:58.809 ERROR (MainThread) [homeassistant.components.automation.reset_schedule_on_sunday] While executing automation automation.reset_schedule_on_sunday
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 551, in async_trigger
    await self.action_script.async_run(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1524, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 409, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 453, in _async_step
    self._handle_exception(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 476, in _handle_exception
    raise exception
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 451, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 684, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1744, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1781, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 770, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 678, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 943, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 709, in _handle_entity_call
    result = hass.async_run_job(partial(getattr(entity, func), **data))  # type: ignore[arg-type]
  File "/usr/src/homeassistant/homeassistant/core.py", line 621, in async_run_job
    return self.async_run_hass_job(HassJob(target), *args)
  File "/usr/src/homeassistant/homeassistant/core.py", line 573, in async_run_hass_job
    hassjob.target(*args)
  File "/config/custom_components/ramses_cc/climate_heat.py", line 215, in svc_reset_system_mode
    self._call_client_api(self._device.reset_mode)
  File "/config/custom_components/ramses_cc/__init__.py", line 216, in _call_client_api
    func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/system/heat.py", line 944, in reset_mode
    return self.set_mode(SYS_MODE_MAP.AUTO_WITH_RESET)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/system/heat.py", line 935, in set_mode
    Command.set_system_mode(self.id, system_mode, until=until)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/protocol/helpers.py", line 27, in wrapper
    return fnc(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/protocol/command.py", line 186, in wrapper
    return _wrapper(fcn, cls, dst_id, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/protocol/command.py", line 163, in _wrapper
    return fcn(cls, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/protocol/command.py", line 900, in set_system_mode
    system_mode = SYS_MODE_MAP._hex(
  File "/usr/local/lib/python3.10/site-packages/ramses_rf/protocol/const.py", line 172, in _hex
    raise KeyError(key)
KeyError: '06'
```

**Comment by zxdavb** [comment]
Thank you - I will add this to my to-do list.

I can reproduce this bug.

**Comment by zxdavb** [comment]
Fixed by commit [471db6e](https://github.com/zxdavb/ramses_rf/commit/471db6e0ebba2e761e5baab041587dc8c459b02e), should be in release **0.22.2**.
